numpy/core/src/umath/reduction.c

```diff
@@ -496,30 +487,29 @@ PyUFunc_ReduceWrapper(PyArrayObject *operand, PyArrayObject *out,
         goto fail;
     }
 
+    /*
```

Comment:
I think this can be moved one if-statement further up, so that it goes together with the sanity check on `wheremask` (and then one can just return `NULL` instead of `goto fail`).

Actually, it might even make more sense in `ufunc_object`, before we even get here (right after getting the identity, which reorderable depends on), so that one doesn't have to pass `reorderable` in at all, but that is perhaps too much of a change, and may be better left for when/if we unify reduction and single-axis ufuncs.

Reply:
Good points